pkg/plugins/resources/helm/utils.go

```diff
@@ -147,12 +150,18 @@ func (c *Chart) MetadataUpdate(chartPath string, dryRun bool) error {
 			md.Version = v.IncMinor().String()
 		case PATCHVERSION:
 			md.Version = v.IncPatch().String()
+		case NOINCREMENT:
+			// Reset Version to its initial value
+			md.Version = oldVersion
```

Comment:
Shouldn't the md.AppVersion also be restored here?

Reply:
No because it's not needed. The chart version can be updated up to three times if spec.semverinc=[major,minor,path,none]
While normally "none" should be used alone, I decided to handle the case where "none" is accidently used with one of the other "rule".
Let me add warning if it's the case

Reply:
+1 for the warning, I would even throw an error from validate()